mlflow/store/artifact/databricks_logged_model_artifact_repo.py

```diff
@@ -0,0 +1,156 @@
+        super().__init__(artifact_uri)
+        m = self._URI_REGEX.search(artifact_uri)
+        if not m:
+            raise MlflowException.invalid_parameter_value(
+                f"Invalid artifact URI: {artifact_uri}. Expected URI of the form "
+                f"databricks/mlflow-tracking/<EXP_ID>/logged_models/<MODEL_ID>"
+            )
+        experiment_id = m.group("experiment_id")
+        model_id = m.group("model_id")
+        relative_path = m.group("relative_path") or ""
+        self.root_path = f"/Mlflow/Artifacts/{experiment_id}/LoggedModels/{model_id}{relative_path}"
+        self.databricks_artifact_repo = DatabricksArtifactRepository(artifact_uri)
+        self.wc = WorkspaceClient(config=Config(enable_experimental_files_api_client=True))
+
+    @property
+    def files_api(self) -> FilesAPI:
+        return self.wc.files
+
+    def _is_dir(self, path: str) -> bool:
+        try:
+            self.files_api.get_directory_metadata(path)
+        except NotFound:
+            return False
+        return True
+
+    def path(self, artifact_path: str) -> str:
+        """
+        Construct the full path to the artifact, given the artifact's relative path.
+        """
+        return f"{self.root_path}/{artifact_path}" if artifact_path else self.root_path
+
+    def _log_artifact(self, local_file: str, artifact_path: Optional[str] = None) -> None:
+        with open(local_file, "rb") as f:
+            self.files_api.upload(self.path(artifact_path), f, overwrite=True)
+
+    def log_artifact(self, local_file: str, artifact_path: Optional[str] = None) -> None:
+        try:
+            self._log_artifact(local_file, artifact_path)
+        except Exception:
+            _logger.info(
+                (
+                    "Failed to log artifact using Databricks SDK, falling back to "
+                    "DatabricksArtifactRepository"
+                ),
+                exc_info=True,
+            )
+            self.databricks_artifact_repo.log_artifact(local_file, artifact_path)
+            return self.databricks_artifact_repo.log_artifact(local_file, artifact_path)
+
+    def _log_artifacts(self, local_dir: str, artifact_path: Optional[str] = None) -> None:
+        futures = []
+        with ThreadPoolExecutor() as executor:
+            for f in Path(local_dir).rglob("*"):
+                if f.is_file():
+                    fut = executor.submit(self.log_artifact, f, artifact_path)
+                    futures.append(fut)
+
+        for fut in futures:
+            fut.result()
+
+    def log_artifacts(self, local_dir, artifact_path=None):
+        try:
+            self._log_artifacts(local_dir, artifact_path)
+        except Exception:
+            _logger.info(
+                (
+                    "Failed to log artifacts using Databricks SDK, falling back to "
+                    "DatabricksArtifactRepository"
+                ),
+                exc_info=True,
```

Comment:
Missing type hints here. According to the repository's typing rules, this should be:

```python
def log_artifacts(self, local_dir: str, artifact_path: Optional[str] = None) -> None:
```